### PR TITLE
doc(tech-lead): new article on tech leads at TELUS digital

### PR DESCRIPTION
**Pull request by kylehodgson**
## Overview

This article defines what tech leads are, and how they work, at TELUS digital

### Features

- create [tech-lead]

---

#### Meta

- [x] provide a descriptive topic
- [x] provide an overview of contribution
- [x] no sensitive content included, such as:
  - security & privacy policy violating content
  - content considered competitive intelligence
  - keys, tokens or credentials
- [x] documentation format follows [this template][template]
- [x] fork is up to date ([see this guide from github](https://help.github.com/articles/syncing-a-fork/))
- [x] "work in progress" commits are squashed ([see "Squashing Commits"](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History))
- [x] commits follow the [Karma][karma-format] or [Commitizen](https://www.npmjs.com/package/commitizen) format

[template]: ../.template.md
[karma-format]: https://karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: process/tech-lead.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Tech Lead Role
+
+## Why
+
+Why do we need Tech Leads in squads and outcome teams?  I like [Pat Kua](https://medium.com/featured-insights/5-tips-for-being-an-effective-tech-lead-7ec36391de34) on the subject:
```
Comment (theetrain) [inline process/tech-lead.md]: Try removing all cases of double spacing after a period.

File: process/tech-lead.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+> **"A good sign of effective tech leadership and a team working together is that everything feels like it’s written by the same developer mindset."**
+
+This is a good thing, as it makes the code much easier to read and understand.
+
+High maturity self organizing teams require only a very light touch from their tech lead, and possibly don't require the role to exist at all - when developers can talk things out and come to alignment on their own efficiently, set good direction and effectively come up with solutions, the tech lead doesn't need to do much. In teams like this, its their responsibility to stay out of the way, to keep the team rolling. On teams that are still working on attaining this level of maturity, its the role of the tech lead to get them there.
```
Comment (theetrain) [inline process/tech-lead.md]: > High maturity `self-organizing` teams...
> ...working on attaining this level of maturity, `it's` the role of the tech lead to get them there.

File: process/tech-lead.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+In our organization today, it can be a little confusing. Sometimes the phrase "Tech Lead" is used to describe someone like this:
+
+* A highly technical leader that acts as a partner to the Outcome Team Product Owner(s). This relationship is sometimes referred to as the "Entrepreneur/Scientist" relationship - both have areas of specialty, both have important skills to bring to the leadership process for all of the squads on the Outcome Team.
+
+It can also be used to describe someone like this:
+
+* A highly technical leader that is closely associated with a particular *squad* - a senior developer that's sitting with the squad they serve daily. They go to their squad's standup every day, they are deeply knowledgeable of their squad's backlog, they act as a partner to the Iteration Manager and/or Business Analyst on the squad. 
+
+For this article, we'll be largely be talking about both - though from time to time when we need to differentiate, we'll refer to the first role as an *Architect*, and the second more squad focused role as a *Tech Lead*.
```
Comment (theetrain) [inline process/tech-lead.md]: > ...we'll `largely be` talking about both
> ...and the second more `squad-focused` role as a _Tech Lead_

File: process/tech-lead.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
...s to spot problems with specific solutions. However, the old saying goes, ["What got you here won't get you there"](https://www.amazon.ca/What-Got-Here-Wont-There/dp/1401301304). In both roles, it can  be tempting to continue being a major contributor to the code base (particularly when the team is challenged to meet deadlines) - though, this is counter productive. It's important for the Tech Lead and Architect to not get in the way of the team. The leader that takes on too much of their teams' work ends up causing a situation where the other developers have less knowledge of the code being written (as it was written by someone else). This can lead to the team feeling paralyzed; they feel like they don't understand sections of the codebase, which leads to them feeling like they shouldn't change them. When leaders come in and out of a code base without fully embedding themselves in the team's processes and mindset, this can be particularly disruptive. Late night coding sessions result in the team scrambling to understand the codebase the next morning, leaders picking up large un-analyzed sections of the code [leads to inefficiency](smaller-stories.md), and causes lots of disruption.
```
Comment (theetrain) [inline process/tech-lead.md]: Double spacing after a period:
> ...a Senior Developer`. This` gives them...

> ...regarding technology `problem-solving`, good...

> ...in both roles, it `can be` tempting

Comment (theetrain) [reply process/tech-lead.md]: For all instances of `code base`, switch to `codebase` to be consistent with other usages in this document.

File: process/tech-lead.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+In the end, unless Architects and Tech Leads are very careful to avoid this kind of disruption, the team will actually scale less and less over time. The rest of the team, without much to do, will either become bored and move on, or their skills will atrophy. Doing software development work is the critical experience it takes to go from developer to Senior, from Senior to Lead. When Leads take on too much of this work, it robs the team of its own chance to become stronger.
```
Comment (theetrain) [inline process/tech-lead.md]: > ...from developer to Senior, `and` from Senior to Lead.

File: process/tech-lead.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+#### Lunch and Learn
+
+A lunch and learn is a format where someone develops some content to present to the team - it can be as simple as a code walkthrough, or more formal, relying on a deck. These tend to be "information sessions", helping a squad learn a new technology all at once. They require a little more planning than Dev Huddles, as someone must come up with the content, but are very effective at getting a squad to a certain base-level of information on something.  Its important not to forget the lunch aspect! Use something like Doordash, or order pizza even. Hungry devs don't code as well as devs who aren't hungry!
```
Comment (theetrain) [inline process/tech-lead.md]: > ...it can be as simple as a code walkthrough, or `a formal presentation that relies on a deck`.

Comment (theetrain) [reply process/tech-lead.md]: Double spacing:
> ...information on something`. It's` important...

File: process/tech-lead.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+#### Mob Coding
+
+[Mob code](https://en.wikipedia.org/wiki/Mob_programming) sessions are a group coding activity. Many teams book a one hour mob code session on a regular basis as an effective way to tackle tech debt, or better understand design patterns. 
+
+### Individual Coaching
+
+It's not uncommon for people in either of these roles to hold [one on one](one-on-ones.md) (1:1) meetings with the other developers working on the teams they serve.  As part of these 1:1 meetings leaders can focus on how a particular developer can continue to improve.
```
Comment (theetrain) [inline process/tech-lead.md]: Double spacing:
> ...teams they serve`. As` part of...

File: process/tech-lead.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+[Mob code](https://en.wikipedia.org/wiki/Mob_programming) sessions are a group coding activity. Many teams book a one hour mob code session on a regular basis as an effective way to tackle tech debt, or better understand design patterns. 
+
+### Individual Coaching
+
+It's not uncommon for people in either of these roles to hold [one on one](one-on-ones.md) (1:1) meetings with the other developers working on the teams they serve.  As part of these 1:1 meetings leaders can focus on how a particular developer can continue to improve. 
+
+## References
+
+https://medium.com/featured-insights/5-tips-for-being-an-effective-tech-lead-7ec36391de34
```
Comment (theetrain) [inline process/tech-lead.md]: Change these links to anchors, and don't add double line-breaks between them:

[5 Tips for being an effective tech lead](https://medium.com/featured-insights/5-tips-for-being-an-effective-tech-lead-7ec36391de34)

File: process/tech-lead.md (hunk 2 above)
Comment (kylehodgson) [reply process/tech-lead.md]: resolved

File: process/tech-lead.md (hunk 3 above)
Comment (kylehodgson) [reply process/tech-lead.md]: resolved

File: process/tech-lead.md (hunk 4 above)
Comment (kylehodgson) [reply process/tech-lead.md]: resolved

File: process/tech-lead.md (hunk 5 above)
Comment (kylehodgson) [reply process/tech-lead.md]: resolved

File: process/tech-lead.md (hunk 6 above)
Comment (kylehodgson) [reply process/tech-lead.md]: resolved

**Comment by kylehodgson** [comment]
This article has been moved to the new confluence wiki on technical leadership. https://telusdigital.atlassian.net/wiki/spaces/TOS/pages/62586886/Technical+Leadership+at+TELUS+digital

**Pull request closed without merging**